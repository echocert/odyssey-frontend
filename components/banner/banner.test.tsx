import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { create } from "react-test-renderer";
import Banner from './banner';


describe('Banner', () => {
  it('renders the Banner Screen', () => {
    render(<Banner />)

    const banner = screen.getByTestId('banner');
    expect(banner).toBeInTheDocument()
  })
  it('check if the Banner Section One is rendered', () => {
    render(<Banner />)

    const banner_section_one = screen.getByTestId('banner-one');
    expect(banner_section_one).toBeInTheDocument()
  })
  it('check if the Banner Section Herotext is rendered', () => {
    render(<Banner />)

    const banner_section_herotext = screen.getByTestId('banner-herotext-one');
    expect(banner_section_herotext).toBeInTheDocument()
  })
  it('check if the Banner Section SubHerotext is rendered', () => {
    render(<Banner />)

    const banner_section_subherotext = screen.getByTestId('banner-subherotext-one');
    expect(banner_section_subherotext).toBeInTheDocument()
  })
  it('check if the Banner Section Description is rendered', () => {
    render(<Banner />)

    const banner_section_description = screen.getByTestId('banner-description');
    expect(banner_section_description).toBeInTheDocument()
  })
  // it('check if the Banner Section Two is rendered', () => {
  //   render(<Banner />)

  //   const banner_section_two = screen.getByTestId('banner-two');
  //   expect(banner_section_two).toBeInTheDocument()
  // })
  it("Verify the snapshot of Banner section is matched", () => {
    const wrap = create(<Banner />).toJSON();
    expect(wrap).toMatchSnapshot();
  });
})