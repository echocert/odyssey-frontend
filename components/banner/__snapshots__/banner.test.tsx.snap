// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner Verify the snapshot of Banner section is matched 1`] = `
[
  <div
    className="banner"
    data-testid="banner"
  >
    <div
      className="banner__section banner__section--one"
      data-testid="banner-one"
    >
      <label
        className="label-hero-l1 label-l1 label-hero"
        data-testid="banner-herotext-one"
        style={
          {
            "color": "#224a96",
            "lineHeight": "5rem",
            "margin": "0",
          }
        }
      >
        Welcome to Odyssey
      </label>
      <label
        className="label-body-l2 label-l2 label-body"
        data-testid="banner-description"
        style={
          {
            "color": "black",
            "lineHeight": "1.5rem",
            "paddingLeft": "0",
          }
        }
      >
        Introducing Odyssey, a revolutionary web app that streamlines the
            certification process including income, asset, and allowance
            modules. Get ready to experience a completely automated journey with
            the Odyssey!
      </label>
    </div>
    <div
      className="banner__section banner__section--two"
      data-testid="banner-two"
    >
      <div
        className="header-logo"
      >
        <img
          alt="Logo"
          src="../assets/images/odyssey_logo_large.png"
        />
      </div>
      <div
        className="header-menu-login"
      >
        <div
          className="btn"
          style={{}}
        >
          <button
            className="buttonEffects btn rounded btn-primary"
            data-testid="login-button"
            onClick={[Function]}
          >
            Login
          </button>
        </div>
      </div>
    </div>
  </div>,
  <div
    style={
      {
        "backgroundImage": "url(/assets/images/home_connected.png)",
        "backgroundSize": "400px 200px",
        "height": "200px",
      }
    }
  />,
]
`;
